dns.go

```diff
@@ -54,7 +54,7 @@ type RR interface {
 	// parse parses an RR from zone file format.
 	//
 	// This will only be called on a new and empty RR type with only the header populated.
-	parse(c *zlexer, origin, file string) *ParseError
+	parse(c *zlexer, origin string) *ParseError
```

Comment:
I think we can just give this a `*ZoneParser` which holds all the info we need (origin, file and the lexer)

Reply:
We could do that. I'm not sure it really gives us much though.